### PR TITLE
HSEARCH-3011 Re-enable building the documentation on all JDKs

### PR DESCRIPTION
**Pull request by Sanne**
- https://hibernate.atlassian.net/browse/HSEARCH-3011
 - https://hibernate.atlassian.net/browse/HSEARCH-3012

### CODE REVIEW

File: documentation/pom.xml [hunk 1]
```diff
@@ -210,7 +210,6 @@
                                         <pdf-style>hibernate</pdf-style>
                                         <pdf-fontsdir>${asciidoctor.aggregated-resources-dir}/theme/fonts</pdf-fontsdir>
                                         <pagenums/>
-                                        <idprefix/>
```
Comment (gsmet) [inline documentation/pom.xml]: I think this change will break all the unqualified anchors which doesn't sound as a good idea.

Is there a reason why you removed it? If there is an issue with Asciidoctor, we should open a bug there.

File: pom.xml [hunk 2]
```diff
@@ -292,10 +293,10 @@
 
         <!-- Asciidoctor -->
         <hibernate-asciidoctor-theme.version>1.0.1.Final</hibernate-asciidoctor-theme.version>
```
Comment (gsmet) [inline pom.xml]: While we're at it, let's update the theme to 1.0.2.Final. It should improve the code formatting in the PDF doc.

File: documentation/pom.xml (hunk 1 above)
Comment (Sanne) [reply documentation/pom.xml]: See the discussions I linked to.. also, this is only PDF so it's not going to break any links.

File: pom.xml (hunk 2 above)
Comment (Sanne) [reply pom.xml]: ok will do, but that's a separate issue.

File: documentation/pom.xml (hunk 1 above)
Comment (gsmet) [reply documentation/pom.xml]: Ah, yes sorry, I didn't have the full context :).

File: pom.xml (hunk 2 above)
Comment (Sanne) [reply pom.xml]: Added a commit for that ;)

**Comment by gsmet** [comment]
I checked the PDF and HTML outputs. Everything looks good, will merge.

**Pull request merged**